File: 18-heaps/2-min-max-heap/main.cpp
// Min and Max Heap - C++ Example

#include <iostream>
#include <queue>
#include <vector>

int main() {
    std::priority_queue<int, std::vector<int>, std::greater<int>> minHeap;
    std::priority_queue<int> maxHeap;

    int nums[] = {5, 3, 8, 1};
    for (int num : nums) {
        minHeap.push(num);
        maxHeap.push(num);
    }

    std::cout << "Min heap order:" << std::endl;
    while (!minHeap.empty()) {
        std::cout << minHeap.top() << std::endl;
        minHeap.pop();
    }

    std::cout << "Max heap order:" << std::endl;
    while (!maxHeap.empty()) {
        std::cout << maxHeap.top() << std::endl;
        maxHeap.pop();
    }
    return 0;
}

File: 18-heaps/2-min-max-heap/main.java
// Min and Max Heap - Java Example

import java.util.Collections;
import java.util.PriorityQueue;

class MinMaxHeapDemo {
    public static void main(String[] args) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>(Collections.reverseOrder());

        int[] nums = {5, 3, 8, 1};
        for (int num : nums) {
            minHeap.add(num);
            maxHeap.add(num);
        }

        System.out.println("Min heap order:");
        while (!minHeap.isEmpty()) {
            System.out.println(minHeap.poll());
        }

        System.out.println("Max heap order:");
        while (!maxHeap.isEmpty()) {
            System.out.println(maxHeap.poll());
        }
    }
}
